#!/usr/bin/env python

PACKAGE = "dcsl_vision_tracker"
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("binary_threshold", int_t, 0, "The cutoff value for thresholding to a binary image", 15, 0, 255)
gen.add("erode_iterations", int_t, 0, "Level of eroding to get rid of small contours", 4, 0, 10)
gen.add("min_blob_size", int_t, 0, "Minimum area of the robot in square pixels", 500, 0, 2000)
gen.add("max_blob_size", int_t, 0, "Maximum area of the robot in the square pixels", 2000, 0, 3000)
gen.add("scale", double_t, 0, "Conversion factor for pixels to real world in 1/pixels", pow(1.45*3.05*1.0/204.0,-1))
gen.add("camera_height", double_t, 0, "Height of camera above water in meters", 3.12)

exit(gen.generate(PACKAGE, "dcsl_beluga_tracker.py", "dcsl_beluga_tracker_config"))
